tests/loops/test_loops.py

```diff
@@ -103,6 +103,36 @@ def test_connect_subloops(tmpdir):
     assert new_batch_loop.trainer is trainer
 
 
+def test_replace_loops():
+    class TestLoop(TrainingEpochLoop):
+        def __init__(self, foo):
+            super().__init__()
+
+    trainer = Trainer(min_steps=123, max_steps=321)
+
+    with pytest.raises(
+        MisconfigurationException, match=r"FitLoop.replace\(TestLoop\)`.*`__init__`.*`TrainingEpochLoop`"
+    ):
+        trainer.fit_loop.replace(TestLoop)
+
+    with pytest.raises(MisconfigurationException, match="Did not find.*same parent class as `PredictionEpochLoop`"):
+        trainer.fit_loop.replace(PredictionEpochLoop)
+
+    class TestLoop(TrainingEpochLoop):
+        ...
+
+    old_loop = trainer.fit_loop.epoch_loop
+    new_loop = trainer.fit_loop.replace(TestLoop)
```

Comment:
Another API option would be to have

```python
trainer.fit_loop.replace({"epoch_loop": TestLoop})
```

with the advantages:
- No need to look up the attribute internally
- Can replace multiple loops at the same time.
- Adds support for loops that have subloops of the same type

And disadvantage:
- Not as cool :P